packages/gatsby-source-contentful/src/plugin-options.js

```diff
@@ -24,40 +25,152 @@ const createPluginConfig = pluginOptions => {
   }
 }
 
-const optionsSchema = Joi.object().keys({
-  accessToken: Joi.string().required().empty(),
-  spaceId: Joi.string().required().empty(),
-  host: Joi.string().empty(),
-  environment: Joi.string().empty(),
-  downloadLocal: Joi.boolean(),
-  localeFilter: Joi.func(),
-  forceFullSync: Joi.boolean(),
-  pageLimit: Joi.number().integer(),
-  proxy: Joi.object().keys({
-    host: Joi.string().required(),
-    port: Joi.number().required(),
-    auth: Joi.object().keys({
-      username: Joi.string(),
-      password: Joi.string(),
-    }),
-  }),
-  useNameForId: Joi.boolean(),
-  // default plugins passed by gatsby
-  plugins: Joi.array(),
-  richText: Joi.object()
-    .keys({
-      resolveFieldLocales: Joi.boolean(),
-    })
-    .default({}),
+const Joi = joi.extend({
+  type: /^s/,
+  rules: {
+    dotenv: {
+      args: [`name`],
+      validate(value, helpers, args) {
+        if (!args.name) {
+          return helpers.error(
+            `any.dotenv requires the environment variable name`
+          )
+        }
+        return value
+      },
+      method(name) {
+        return this.$_addRule({ name: `dotenv`, args: { name } })
+      },
+    },
+  },
```

Comment:
Do we want to put this in Joi? What happens if plugins use the same env var? why can't people just put I n process.env.x in gatsby-config? If we expand to cli/admin we could go for `env:x` so in the example below I would put in `env:CONTENTFUL_SPACE_ID`

Reply:
If CLI/Admin know a value is supposed to go into dotenv, we can write both `process.env.CONTENTFUL_SPACE_ID` into gatsby-config.js as well as the actual value into the `.env` file.

If this isn't marked in the schema, users might enter the the secret value into the input and we'd write it in the `gatsby-config.js` without the user realising, which is not good. :blush: